### PR TITLE
chore: minor clean-up of `math/base/special/acsc`

### LINKED ISSUE #859: [RFC]: Add C implementation for `@stdlib/math/base/special/acsc`
### Description

This RFC proposes adding native C implementation for [`@stdlib/math/base/special/acsc`](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/acsc/).

```c
double stdlib_base_acsc( const double x );
```

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Pranavchiku**
Resolves #859.

## Checklist

- [x] update readme.md
- [x] include.gypi
- [x] binding.gyp
- [x] include/stdlib/math/base/special/
- [x] src
- [x] manifest.json
- [x] lib
- [x] examples
- [x] benchmark
- [x] test
* * *

Feels good to be back :)

### CODE REVIEW

**Comment by Pranavchiku** [comment]
Not sure how can I run benchmarks and tests, I followed previous commands like

```console
% make benchmark-c BENCHMARKS_FILTER=.*/math/base/special/acsc/.*                                            
zsh: no matches found: BENCHMARKS_FILTER=.*/math/base/special/acsc/.*
```

```console
% make examples-c EXAMPLES_FILTER=.*/math/base/special/acsc.*                                               
zsh: no matches found: EXAMPLES_FILTER=.*/math/base/special/acsc/.*
```

How may I run javascript tests?

**Comment by kgryte** [comment]
```
make test TESTS_FILTER=".*/math/base/special/acsc/.*"
```

should work.

File: lib/node_modules/@stdlib/math/base/special/acsc/benchmark/c/benchmark.c [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+	double elapsed;
+	double x;
+	double y;
+	double t;
+	int i;
+
+	t = tic();
+	for ( i = 0; i < ITERATIONS; i++ ) {
+		x = ( 2.0*rand_double() ) + 1.0;
+		y = acsc( x );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acsc/benchmark/c/benchmark.c]: I do not believe that `acsc` is part of C99 `math.h`: https://en.cppreference.com/w/c/numeric/math.

As such, TMK, there is no "built-in" implementation to measure against and this file can be removed.

File: lib/node_modules/@stdlib/math/base/special/acsc/include/stdlib/math/base/special/acsc.h [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+/*
+* If C++, prevent name mangling so that the compiler emits a binary file having undecorated names, thus mirroring the behavior of a C compiler.
+*/
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+* Computes the inverse cotangent of a double-precision floating-point number.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acsc/include/stdlib/math/base/special/acsc.h]:
```suggestion
* Computes the arccosecant of a double-precision floating-point number.
```

File: lib/node_modules/@stdlib/math/base/special/acsc/lib/native.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+// MODULES //
+
+var addon = require( './../src/addon.node' );
+
+
+// MAIN //
+
+/**
+* Computes the inverse cosecant of a double-precision floating-point number.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acsc/lib/native.js]:
```suggestion
* Computes the arccosecant of a double-precision floating-point number.
```

File: lib/node_modules/@stdlib/math/base/special/acsc/lib/native.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+*
+* @private
+* @param {number} x - input value
+* @returns {number} inverse cosecant (in radians)
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acsc/lib/native.js]:
```suggestion
* @returns {number} arccosecant (in radians)
```

File: lib/node_modules/@stdlib/math/base/special/acsc/package.json [hunk 5]
```diff
@@ -18,7 +18,9 @@
     "benchmark": "./benchmark",
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acsc/package.json]: You need to add a `gypfile: true` field after L16.

File: lib/node_modules/@stdlib/math/base/special/acsc/src/main.c [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/special/acsc.h"
+#include "stdlib/math/base/special/asin.h"
+
+/**
+* Computes the inverse cosecant of a double-precision floating-point number.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acsc/src/main.c]:
```suggestion
* Computes the arccosecant of a double-precision floating-point number.
```

File: lib/node_modules/@stdlib/math/base/special/acsc/src/main.c [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+*
+* @param x    input value
+* @return	  output value
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acsc/src/main.c]:
```suggestion
* @return      output value
```

File: lib/node_modules/@stdlib/math/base/special/acsc/test/test.native.js [hunk 8]
```diff
@@ -0,0 +1,132 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2022 The Stdlib Authors.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acsc/test/test.native.js]:
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/acsc/test/test.native.js [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+	t.end();
+});
+
+tape( 'the function returns `NaN` if provided `NaN`', opts, function test( t ) {
+	var v = acsc( NaN );
+	t.equal( isnan( v ), true, 'returns NaN' );
+	t.end();
+});
+
+tape( 'the function returns `NaN` if provided a nonpositive value greater than `-1`', function test( t ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acsc/test/test.native.js]:
```suggestion
tape( 'the function returns `NaN` if provided a nonpositive value greater than `-1`', opts, function test( t ) {
```

File: lib/node_modules/@stdlib/math/base/special/acsc/src/main.c [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+*/
+
+#include "stdlib/math/base/special/acsc.h"
+#include "stdlib/math/base/special/asin.h"
+
+/**
+* Computes the arccosecant of a double-precision floating-point number.
+*
+* @param x    input value
+* @return      output value
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acsc/src/main.c]:
```suggestion
* @return     output value
```

**Comment by kgryte** [comment]
@Pranavchiku I am not sure why those commands are not working for you. Can you trying surrounding the filter in quotes?

```
make benchmark-c BENCHMARKS_FILTER=".*/math/base/special/acsc/.*"
```

**Comment by kgryte** [comment]
@Pranavchiku This is looking good. Thanks for working on this. The builtin C benchmark can be removed as `acsc` is not a part of C99, TMK. We also need to add a `gypfile` field to the `package.json`. Once resolved, this PR should be ready for merge.

**Comment by kgryte** [comment]
@Pranavchiku Are you interested in working to get this PR over the finish line?

**Comment by Pranavchiku** [comment]
I am afraid I will not be able to do it atm. I am fine if anyone else pick it up or if it is not a blocker I can look it up later at end of June.

**Comment by Planeshifter** [comment]
We already merged a [PR for the C implementation ](https://github.com/stdlib-js/stdlib/pull/1887). Merged latest changes from `develop` into the PR branch and will merge the now much smaller PR, which just consists of a few documentation and style improvements, shortly.

**Review by Planeshifter — APPROVED** [review]

**Pull request merged**